Sources/App/CRP/CRPProcess.swift

```diff
@@ -0,0 +1,122 @@
+import Foundation
+import Vapor
+import Stevenson
+
+enum CRPProcess {
+    enum Option: String {
+        case repo
+        case branch
+        case skipTicket
+        case skipFixVersion
+
+        func get<T: Decodable>(from request: Request) throws -> T {
+            guard let result = request.query[T.self, at: self.rawValue] else {
+                throw CRPProcess.Error.missingParameter(self.rawValue)
+            }
+            return result
+        }
+    }
+
+    enum Error: Swift.Error {
+        case invalidParameter(key: String, value: String, expected: String)
+        case missingParameter(String)
+    }
+
+    static func apiRequest(request: Request, github: GitHubService, jira: JiraService) throws -> Future<Response> {
+        let repo: String = try Option.repo.get(from: request)
```

Comment:
doesn't Vapor provide any builtin way to access query parameters?

Reply:
Also I was expecting this to redirect to existing CRP method using with slack command, but the code is not moved/removed. Is it not possible?

Reply:
It does, that's what is used by the implementation of my `Option.get` helper a few lines above: `request.query.get(…)` / `request.query[T.Type, at: …]`. The helper just allows me to avoid duplicating the access to `.rawValue` and the `throws CRPProcess.Error` each time.

Reply:
The only things that are common are already factored. As detailed in the PR description:

> Everything is implemented in the CRPProcess namespace. the sequence of function calls is nearly the same as the functions called from SlackCommand+CRP (github.changelog() + jira.create(issue:) + jira.createAndSetFixedVersions()) but params parsing, optionality of checks and report as JSON is what makes it different.

Only things I see we could have factored a bit more is combine `JiraService.makeCRPIssue` + `jira.create(issue:)` in a convenience helper method. (Be we wouldn't gain much from that imho as `jira.create` is a one-liner done after `makeCRPIssue`, and the rest differs in what we `map` because in one case we have optionals and the other not)